syntax = "proto3";
package ek;

message ExpertForwardRequest {
  int32 layer = 1;
  int32 idx = 2;
  int32 batch_size = 3;
  bytes tensor = 4;
}

message ExpertForwardReply {
  bytes output_tensor = 1;
}

service ExpertComputation {
  rpc Forward(ExpertForwardRequest) returns (ExpertForwardReply);
}
